bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -76,7 +87,47 @@ async def cog_load(self) -> None:
 
             self.scheduler.schedule_at(next_reschedule_point, -1, self.cog_load())
 
-        log.trace("Done rescheduling")
+        log.trace("Done rescheduling expirations, scheduling tidy up tasks.")
+
+        for key, expire_at in await self.messages_to_tidy.items():
+            channel_id, message_id = map(int, key.split(":"))
+            expire_at = dateutil.parser.isoparse(expire_at)
+
+            log.trace(
+                "Scheduling tidy up for message %s in channel %s at %s",
+                message_id, channel_id, expire_at
+            )
+
+            self.tidy_up_scheduler.schedule_at(
+                expire_at,
+                message_id,
+                self._delete_infraction_message(channel_id, message_id)
+            )
+
+    async def _delete_infraction_message(
+        self,
+        channel_id: int,
+        message_id: int
+    ) -> None:
+        """
+        Delete a message in the given channel.
+
+        This is used to delete infraction messages after a certain period of time.
+        """
+        try:
+            channel = await get_or_fetch_channel(self.bot, channel_id)
+            message = await channel.fetch_message(message_id)
+            await message.delete()
+            log.trace(f"Deleted infraction message {message_id} in channel {channel_id}.")
+        except discord.NotFound:
+            log.warning(f"Channel or message {message_id} not found in channel {channel_id}.")
+        except discord.Forbidden:
+            log.warning(f"Bot lacks permissions to delete message {message_id} in channel {channel_id}.")
```

Comment:
Unsure if this is a warning, since trying to delete messages in closed threads will also raise Forbidden iirc

Reply:
Ah I hadn't realised that manage messages does not override this. I'll drop it to an info log.